handlers.go

```diff
@@ -247,6 +247,12 @@ func (r *resolver) allowCharmUpgrade(existingApp *Application, bundleApp *charm.
 	if existingApp.Revision == -1 {
 		return false, nil
 	}
+	// This handles the case that the existing application doesn't have a
+	// channel, so we're talking to an older controller. In that situation the
+	// older path way would ignore channel upgrades.
+	if existingApp.Channel == "" {
```

Comment:
So we've got the model from an older controller. Why shouldn't we be allowed to do a channel upgrade if we want to given that it is the updated client that handles the changes anyway? Should we instead error out and allow a `--force` here?

Reply:
Keeping consistency with older controllers. We can revisit this later if we want to?

Reply:
But if I want to upgrade using a bundle with a specific charm channel, wouldn't an _older_ juju client let me go ahead anyway if I specify the revision (charmstore) + a channel?

Reply:
Resolved